http_101.rst

```diff
@@ -1,20 +1,369 @@
+TRACE         Perform a message loop-back test along the path to the target resource.
+============= ===============
+
+(source: :rfc:`7231`)
+
+After the empty line, an eventual body message contains the payload (data) of the request.
+
+Response
+========
+
+The server parses the response, and, depending on the validity of the request, sends one or more responses.
+An HTTP response starts with the status line, which contains the protocol version, succes or error code, and a reason message.
+
+The status line is mostly followed by a header section containing server information and other metadata.
+An empty line indicates the end of the status line and header.
+The final part of the response, the message body, contains the payload send back to the client.
+
+.. code-block:: console
+
+	HTTP/1.1 200 OK
+	Date: Wed, 20 May 2015 19:45:37 GMT
+	Server: Apache/2.2.22 (Debian)
+	Last-Modified: Tue, 19 May 2015 14:25:26 GMT
+	ETag: "9dac8-2d-5178e2f1c2e8b"
+	Accept-Ranges: bytes
+	Content-Length: 45
+	Vary: Accept-Encoding
+	Content-Type: text/html
+
+	<html><body><h1>It works!</h1></body></html>
+
+Common response codes
+---------------------
+
+============= =====================  ====================
+Code	      Reason		     Description
+============= =====================  ====================
+200	      OK	             The request has succeeded.
+301	      Moved Permenantly      The target has moved to a new location and future references should use a new location. The server should mention the new location in the ``Location`` header in the response. 
+302	      Found		     The target resources been temporarily moved. As the move is temporarily, future references should still use the same location.
+400	      Bad Request	     The server cannot process the request as it is perceived as invalid.
+401	      Unauthorized	     The request to the target resource is not allowed due to missing or incorrect authentication credentials.
+403	      Forbidden		     The request to the target resource is not allowed for reasons unrelated to authentication. 
+404	      Not Found		     The target resource was not found.
+500	      Internal Server Error  The server encountered an unexpected error.
+502	      Bad Gateway            The server is acting as a gateway/proxy and received an invalid response from a server it contacted to fulfill the request.
+503	      Service Unavailable    The server is currently unable to fulfill the request.
+============= =====================  ====================
+
+
+
 Tools: Speaking http with telnet/netcat/curl
 ============================================
 
-Understanding how a browser/tool parses a url and server receives it for a
-request
+Telnet
+------
+
+The telnet utility is used for interactive communication to a host on a given port. 
+Once the connection to the remote host is established, an HTTP request can be send to the host by typing it in the prompt:
+
+.. code-block:: console
+
+	$ telnet apache.org 80
```

Comment:
Could you possibly try examples with opsschool.org? Could help readers try this out.